django/contrib/admin/views/main.py

```diff
@@ -133,9 +133,17 @@ def get_filters(self, request):
                 spec = list_filter(request, lookup_params, self.model, self.model_admin)
             else:
                 field_path = None
+                title = None
                 if isinstance(list_filter, (tuple, list)):
-                    # This is a custom FieldListFilter class for a given field.
-                    field, field_list_filter_class = list_filter
+                    field, second_param = list_filter
+
+                    if isinstance(second_param, str):
+                        # Second parameter is an override for the default title
+                        field_list_filter_class = FieldListFilter.create
+                        title = second_param
+                    else:
+                        # This is a custom FieldListFilter class for a given field.
+                        field_list_filter_class = second_param
```

Comment:
So it feels odd that you can provide a `title` *or* a custom `FieldListFilter` class, but not both. I think it should be possible to provide a 3-tuple to allow `(field, class, title)`. The disadvantage is that is it not obvious given `(field, class)` and `(field, title)` whether it ought to be `(field, class, title)` or `(field, title, class)`, so...

...perhaps a dict with keys `field`, `class` and `title` would be better instead as it would allow for further additional options that might be required in the future. We could keep the 2-tuple with the previous behaviour for backward compatibility.

Perhaps you could take this to the [mailing list](https://groups.google.com/forum/#!forum/django-developers) to get approval for a sane API for this?

Reply:
Thanks for the review!

It might seem unintuitive at first to provide either `title` or a custom `FieldListFilter` since these are two very different things, but I guess it makes sense. If you don't like the default name of a field (which is the `verbose_name`) you pass a title along with it in a tuple `(field, title)`. If you have created a custom `FieldListFilter` then you can define the title with the class attribute `title` of your custom `FieldListFilter` and add the tuple `(field, class)` to `list_filter`. Since you have this class attribute I hardly see a case where 3-tuple `(field, class, title)` would be necessary. Maybe if you have one custom filter that you want to use multiple times with different titles.

Dictionaries are more verbose than tuples to define, but of course, there is a point where things get more complex and they are justified by better readability and clarity. Keeping the existing options with 2-tuples and adding the option to supply a dict is also an option.

I will take this discussion to the mailing list and let see if we can find a consensus there.

Reply:
Yes a custom class can have the title hard coded, but if creating a reusable custom list filter it would be nice to allow changing title in the same way.

Also, Django has some built-in filter classes such as `RelatedOnlyFieldListFilter` and the upcoming `EmptyFieldListFilter`. Surely we should be able to set the title when using these?

Reply:
To me the idea of this ticket is to give developers an easy way of changing the title _when this is all they want to do_. i.e. when they are otherwise happy with the default filter. If they wanted to use a different filter class, then they automatically know which class that is and can easily create a little subclass to set the title there.